tests/integration/test_charm.py

```diff
@@ -1,49 +1,137 @@
 # Copyright 2023 Canonical Ltd.
 # See LICENSE file for licensing details.
 
-import pytest
-from ops.model import ActiveStatus
+import json
+import logging
+
+import ops
+import requests
+from boto3 import client
+from botocore.config import Config
+from ops.model import Application
 from pytest_operator.plugin import OpsTest
 
+logger = logging.getLogger(__name__)
+
+
+async def test_workload_online_default(
+    ops_test: OpsTest,
+    app: Application,
+):
+    unit_informations = json.loads(
+        (await ops_test.juju("show-unit", app.units[0].name, "--format", "json"))[1]
+    )
+    mattermost_ip = unit_informations[app.units[0].name]["address"]
+    response = requests.get(f"http://{mattermost_ip}:8065", timeout=5)
+    assert response.status_code == 200
+    assert "Mattermost" in response.text
+
+
+async def test_s3_storage(
+    ops_test: OpsTest,
+    model: ops.model.Model,
+    app: Application,
+    localstack_s3_config: dict,
+    test_user: dict,
+    tmp_path,
```

Comment:
```suggestion
localstack_s3_config: dict[str,str],
test_user: dict[str,str],
tmp_path: pathlib.Path,
```
Minor suggestion for type hinting :)

Reply:
type hinting added